tests/basic/execution.rs

```diff
@@ -244,3 +245,37 @@ fn failure_outside_execution() {
     let lock = shuttle::sync::Mutex::new(0u64);
     let _ = lock.lock().unwrap();
 }
+
+fn reset_step_bound(do_reset: bool) {
+    let mut config = Config::new();
+    config.max_steps = MaxSteps::FailAfter(20);
```

Comment:
Not absolutely necessary, but I'm wondering if 20 is tight here. Would be interesting to write this as a formula that has the number of threads and yields as parameters.

Reply:
> I'm wondering if 20 is tight here?

What does tight mean? As small as possible?

We know that if you run the test without resetting step counts, then it will panic, and if you do reset, then it won't. That sounds like a good enough test to me, and any number other than 20 which satisfies that (one test failing, one test succeeding) is equivalent in its value.

> Would be interesting to write this as a formula that has the number of threads and yields as parameters.

Sounds like a waste of time.

Reply:
Yeah, as small as possible. Basically the example shows that a user needs to think about the worst case, where all tasks are right before resetting the step count. Could have another test with `do_reset = true` and `max_steps` set to the smallest successful number - 1, which should panic.

Reply:
Ack, added